test/core/util/resolve_localhost_ip46.cc

```diff
@@ -59,4 +59,20 @@ void LocalhostResolves(bool* ipv4, bool* ipv6) {
   *ipv6 = localhost_to_ipv6;
 }
 
+bool RunningWithIPv6Only() {
+  bool localhost_resolves_to_ipv4 = false;
+  bool localhost_resolves_to_ipv6 = false;
+  LocalhostResolves(&localhost_resolves_to_ipv4, &localhost_resolves_to_ipv6);
+  return !localhost_resolves_to_ipv4 && localhost_resolves_to_ipv6;
+}
+
+absl::string_view LocalIp() {
```

Comment:
Almost all calls to this method are as a part of string cancatenation:
```
absl::StrCat(grpc_core::LocalIp(), ":", backends_[0]->port())
```
I would suggest considering having that in the utility function.